src/com/palmergames/bukkit/towny/TownyAsciiMap.java

```diff
@@ -70,66 +79,87 @@ public static void generateAndSend(Towny plugin, Player player, int lineHeight)
 						if (townblock.getType().equals(TownBlockType.COMMERCIAL))
-							townyMap[y][x] = Colors.Blue;
-						townyMap[y][x] += "$";
+							townyMap[y][x] = townyMap[y][x].color(NamedTextColor.BLUE);
+						townyMap[y][x] = townyMap[y][x].content("$");
 					} else if (townblock.isHomeBlock())
-						townyMap[y][x] += "H";
+						townyMap[y][x] = townyMap[y][x].content("H");
 					else
-						townyMap[y][x] += townblock.getType().getAsciiMapKey();
+						townyMap[y][x] = townyMap[y][x].content(townblock.getType().getAsciiMapKey());
+					
+					TownyObject owner = townblock.getTown();
+					if (townblock.hasResident())
+						owner = townblock.getResident();
+					
+					TextComponent forSaleComponent = Component.empty();
+					if (TownyEconomyHandler.isActive() && townblock.getPlotPrice() != -1)
+						forSaleComponent = Component.text(String.format(ChunkNotification.forSaleNotificationFormat, TownyEconomyHandler.getFormattedBalance(townblock.getPlotPrice())).replaceAll("[\\[\\]]", "")).color(NamedTextColor.YELLOW).append(Component.newline());
+
+					TextComponent hoverComponent = Component.text(Translation.of("status_town") + townblock.getTown().getName() + (townblock.hasResident() ? " (" + townblock.getResident().getName() + ")" : "")).color(NamedTextColor.GREEN).append(Component.text(" (" + tby + ", " + tbx + ")").color(NamedTextColor.WHITE)).append(Component.newline())
+						.append(Component.text(Translation.of("status_plot_type")).color(NamedTextColor.GREEN)).append(Component.text(townblock.getType().getName()).color(NamedTextColor.GREEN)).append(Component.newline())
+						.append(forSaleComponent)
+						.append(Component.text(Translation.of("status_perm") + ((owner instanceof Resident) ? townblock.getPermissions().getColourString().replace("n", "t") : townblock.getPermissions().getColourString().replace("f", "r")))).append(Component.newline())
+						.append(Component.text(Translation.of("status_perm") + ((owner instanceof Resident) ? townblock.getPermissions().getColourString2().replace("n", "t") : townblock.getPermissions().getColourString2().replace("f", "r")))).append(Component.newline())
+						.append(Component.text(Translation.of("status_pvp") + ((!CombatUtil.preventPvP(world, townblock)) ? Translation.of("status_on"): Translation.of("status_off")) + 
+							Translation.of("explosions") + ((world.isForceExpl() || townblock.getPermissions().explosion) ? Translation.of("status_on"): Translation.of("status_off")) + 
+							Translation.of("firespread") + ((townblock.getTown().isFire() || world.isForceFire() || townblock.getPermissions().fire) ? Translation.of("status_on"):Translation.of("status_off")) + 
+							Translation.of("mobspawns") + ((world.isForceTownMobs() || townblock.getPermissions().mobs) ?  Translation.of("status_on"): Translation.of("status_off"))));
+
+					townyMap[y][x] = townyMap[y][x].hoverEvent(HoverEvent.showText(hoverComponent)).clickEvent(ClickEvent.runCommand("/towny:plot perm " + tby + " " + tbx));
```

Comment:
So this is allowing the player to click on the townblock and run the /plot perm command for it, but nearly all of the things that plot perm will return are already on-screen in the hover text. 

I'm wondering if including the perm lines in the hovertext might be overkill. CombatUtil.preventPVP() for instance will be firing a PVP-status event for every single townblock on screen.

Reply:
A compromise could be to add a line to the hovertext "Click to see detailed information."